share/picongpu/benchmarks/TWEAC-FOM/include/picongpu/param/incidentField.param

```diff
@@ -67,523 +69,180 @@
+                          Params::PULSE_LENGTH_SI,
+                          Params::W0_SI,
+                          -Params::PHI,
+                          Params::BETA_0,
+                          Params::TDELAY,
+                          Params::AUTO_TDELAY,
+                          Params::Polarization)
                 {
                 }
 
                 HDINLINE float3_X operator()(const floatD_X& totalCellIdx) const
                 {
-                    constexpr float_64 WAVE_LENGTH_SI = 0.8e-6;
-                    constexpr float_64 UNITCONV_A0_to_Amplitude_SI = -2.0 * PI / WAVE_LENGTH_SI
-                        * sim.si.getElectronMass() * sim.si.getSpeedOfLight() * sim.si.getSpeedOfLight()
-                        / sim.si.getElectronCharge();
-                    const float_64 _A0 = 3.25 * 0.01; // reduced for FOM benchmark
-                    const float3_64 invUnitField
+                    float3_64 const invUnitField
                         = float3_64(1.0 / m_unitField[0], 1.0 / m_unitField[1], 1.0 / m_unitField[2]);
-                    const float3_X amplitude
-                        = precisionCast<float_X>(float_64(_A0 * UNITCONV_A0_to_Amplitude_SI) * invUnitField);
-                    return float3_X::create(-0.7057829460593135) * amplitude
-                        * twtsFieldE2(totalCellIdx, m_currentStep);
+                    float3_X const amplitude = precisionCast<float_X>(Params::AMPLITUDE_SI * invUnitField);
+                    return amplitude * twtsFieldEneg(totalCellIdx, m_currentStep);
                 }
 
 #if PARAM_COMPONENTWISE
                 template<uint32_t T_component>
                 HDINLINE float_X getComponent(floatD_X const& totalCellIdx) const
                 {
-                    constexpr float_64 WAVE_LENGTH_SI = 0.8e-6;
-                    constexpr float_64 UNITCONV_A0_to_Amplitude_SI = -2.0 * PI / WAVE_LENGTH_SI
-                        * sim.si.getElectronMass() * sim.si.getSpeedOfLight() * sim.si.getSpeedOfLight()
-                        / sim.si.getElectronCharge();
-                    const float_64 _A0 = 3.25 * 0.01; // reduced for FOM benchmark
-                    const float_64 invUnitField = 1.0 / m_unitField[T_component];
-                    const float_X amplitude
-                        = precisionCast<float_X>(float_64(_A0 * UNITCONV_A0_to_Amplitude_SI) * invUnitField);
-                    return -0.7057829460593135_X * amplitude
-                        * twtsFieldE2.getComponent<T_component>(totalCellIdx, m_currentStep);
+                    float_64 const invUnitField = 1.0 / m_unitField[T_component];
+                    float_X const amplitude = precisionCast<float_X>(Params::AMPLITUDE_SI * invUnitField);
+                    return amplitude * twtsFieldEneg.getComponent<T_component>(totalCellIdx, m_currentStep);
                 }
 #endif
             };
 
-            class FunctorE3
+            class FunctorBpos
```

Comment:
We could also use the automatic B-Field computation via SVEA in the incidentField method. Or would this not be precise enough?

Reply:
Sure, but this still needs to be tested for precision and compared to the variant with both E- and B-field.